src/compare.rs

```diff
@@ -0,0 +1,40 @@
+use std::{convert::TryInto, time::Instant};
+
+use crate::{Datadog, TagsProvider};
+
+const EXPERIMENTS_METRIC_NAME: &str = "experiments";
+
+pub struct TimingGuard {
+    start: Instant,
+    tags: Vec<String>,
+}
+
+impl TimingGuard {
+    pub(crate) fn new<S: AsRef<str>>(name: impl AsRef<str>, path: usize, tags: impl TagsProvider<S>) -> Self {
+        let mut res = Self {
+            start: Instant::now(),
+            tags: tags.as_ref().iter().map(|t| t.as_ref().to_string()).collect(),
+        };
+        res.tags.push(format!("path_taken:{}", path));
+        // All comparisons go under an "experiment" metric, and then we tag each comparison with the name of the experiment
+        // This is to make it easy to browse all the different experiments in datadog by looking at the metric information
+        res.tags.push(format!("name:{}", name.as_ref()));
+        res
+    }
+}
+
+impl Drop for TimingGuard {
```

Comment:
Could we have a method to leave the guard/pause the timer without dropping the guard?

Reply:
I can add, but can you give me a use case? The intended usage here is to time how long blocks of code take to execute, and even in the presence of e.g. an async network call, the time spent awaiting that network call is part of the time spent executing the code block

Reply:
makes sense